includes/class-wc-payments.php

```diff
@@ -231,7 +231,7 @@ public static function init() {
 		}
 
 		// Payment Request and Apple Pay.
-		self::$payment_request_button_handler = new WC_Payments_Payment_Request_Button_Handler( self::$account );
+		self::$payment_request_button_handler = new WC_Payments_Payment_Request_Button_Handler( self::$account, self::$card_gateway );
```

Comment:
Injecting `WC_Payment_Gateway_WCPay` so that it's easier to test

Reply:
I was thinking it maybe better to use the getter `WC_Payments::get_gateway()` to get the gateway instance. Although the method does exact same thing at the moment, but any future logic encapsulated in the `get_gateway()` method  will be included.

That said, the first parameter is using `self::$account` which could also be replaced by `self::get_account_service()`. Given that the other places in this constructor fetch static variables directly, it is probably better to be consistent. This is fine.

Reply:
Oh yeah, that's a good point - it looks like `self::$account`, `self::$card_gateway`, etc. are [directly referenced in the `WC_Payments` class in multiple places](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L203-L266) - looks like it could be a good candidate for a refactor when passing the objects through DI in the constructors, when the classes are instantiated 👍